tests/compilation_cache_test.py

```diff
@@ -0,0 +1,120 @@
+    hash_obj2 = self.create_sudo_hash_obj(cc._hash_executable_build_options, compile_options_filled.executable_build_options)
+    hash_obj3 = self.create_sudo_hash_obj(cc._hash_executable_build_options, compile_options_not_filled.executable_build_options)
+    self.assertEqual(hash_obj1.hex(), hash_obj2.hex())
+    self.assertNotEqual(hash_obj1.hex(), hash_obj3.hex())
+
+  def test_debug_options(self):
+    compile_options = jax.lib.xla_bridge.get_compile_options(
+                      num_replicas=1, num_partitions=1)
+    hash_obj1 = self.create_sudo_hash_obj(cc._hash_debug_options, compile_options.executable_build_options.debug_options)
+    hash_obj2 = self.create_sudo_hash_obj(cc._hash_debug_options, compile_options.executable_build_options.debug_options)
+    self.assertEqual(hash_obj1.hex(), hash_obj2.hex())
+
+  def test_libtpu(self):
+    hash_obj1 = self.create_sudo_hash_obj(cc._hash_libtpu, jax.lib.xla_bridge.get_backend())
+    hash_obj2 = self.create_sudo_hash_obj(cc._hash_libtpu, jax.lib.xla_bridge.get_backend())
+    self.assertEqual(hash_obj1.hex(), hash_obj2.hex())
+
+  def tests_hash_int(self):
+    hash_obj1 = self.create_sudo_hash_obj(cc._hash_int, 90)
+    hash_obj2 = self.create_sudo_hash_obj(cc._hash_int, 8)
+    hash_obj3 = hashlib.sha256()
+    hash_obj3.update((8).to_bytes(8, byteorder='big'))
+    self.assertEqual(hash_obj2, hash_obj3.digest())
+    self.assertNotEqual(hash_obj1, hash_obj2)
+
+  def tests_hash_bool(self):
+    hash_obj1 = self.create_sudo_hash_obj(cc._hash_bool, False)
+    hash_obj2 = self.create_sudo_hash_obj(cc._hash_bool, True)
+    hash_obj3 = hashlib.sha256()
+    hash_obj3.update((True).to_bytes(1, byteorder='big'))
+    self.assertEqual(hash_obj2, hash_obj3.digest())
+    self.assertNotEqual(hash_obj1, hash_obj2)
+
+  def tests_hash_string(self):
+    hash_obj1 = self.create_sudo_hash_obj(cc._hash_string, "foo")
+    hash_obj2 = self.create_sudo_hash_obj(cc._hash_string, "bar")
+    hash_obj3 = hashlib.sha256()
+    hash_obj3.update(("bar").encode('utf-8').strip())
+    self.assertEqual(hash_obj2, hash_obj3.digest())
+    self.assertNotEqual(hash_obj1, hash_obj2)
+
+  def test_same_hash_key(self):
+    computation = jax.xla_computation(lambda x, y: x + y)(1, 1)
+    compile_options = jax.lib.xla_bridge.get_compile_options(
+                       num_replicas=1, num_partitions=1)
+    self.assertEqual(cc.get_cache_key(computation, compile_options),
+                     cc.get_cache_key(computation, compile_options))
+
+  def tets_different_hash_key(self):
+    computation = jax.xla_computation(lambda x, y: x + y)(1, 1)
+    compile_options_not_filled = jax.lib.xla_bridge.get_compile_options(
+                       num_replicas=1, num_partitions=1)
+    compile_options_filled = self.fill_compile_options()
+    self.assertNotEqual(cc.get_cache_key(computation, compile_options_not_filled),
+                        cc.get_cache_key(computation, compile_options_filled))
```

Comment:
Also test that two different computations with the same compile options produce different hashes